File: Etapa3-Construcao/3.2-Stack2_Java_Spring/src/main/java/com/hotel/quartos/domain/CamaQuarto.java
package com.hotel.quartos.domain;

import jakarta.persistence.*;
import lombok.*;

// Entidade para suportar N camas por quarto (OneToMany Quarto->CamaQuarto)
@Entity
@Table(name = "cama_quarto")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CamaQuarto {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoCama tipoCama;
}

File: Etapa3-Construcao/3.1-Stack1_Node_TS/src/domain/entities/CamaQuarto.ts
import { TipoCama } from '../enums/TipoCama.js';

// Entidade de associação para suportar N camas por quarto.
export class CamaQuarto {
  // Em produção poderia ser um UUID; aqui mantemos simples.
  constructor(public readonly tipoCama: TipoCama) {}
}
